app/models/piece.rb

```diff
@@ -73,4 +73,17 @@ def move_to!(new_x, new_y)
       self.update(x_position: new_x, y_position: new_y)
     end
   end
+
+  
+
+  def valid_move?
```

Comment:
I do not think we should have this method here. The rspec tests will recognize the King model since you setup the factory in factories.rb correctly.

Reply:
I think I understand why this was here now. But we have changed the is_obstructed? method so for now I think this should be removed. We will most likely add a valid_move? method to the Pieces.rb model later on.